hal/i2c_api.h

```diff
@@ -61,91 +76,170 @@ typedef struct i2c_s i2c_t;
+ *  the contents must not be modified. The value of the specified `address` is
+ *  ignored when configured in slave mode, in master mode it contains the
+ *  address of the target peripheral. This is a 7-bit value unless 10-bit
+ *  addressing is configured and supported by the target.
+ *
+ *  When in master mode the operation consists of:
+ *   - Address the slave as a Master transmitter.
+ *   - Transmit data to the addressed slave.
+ *   - Generate a STOP condition if the specified `stop` field is true.
  *
  *  @param obj     The I2C object
  *  @param address 7-bit address (last bit is 0)
  *  @param data    The buffer for sending
  *  @param length  Number of bytes to write
- *  @param stop    Stop to be generated after the transfer is done
+ *  @param stop    If true, stop will be generated after the transfer is done
+ *
+ *  @note If the current platform supports multimaster operation the transfer
+ *        will block until the peripheral can gain arbitration of the bus and
+ *        complete the transfer. If the device does not support multimaster
+ *        operation this function is not safe to execute when the bus is shared
+ *        with another device in master mode.
+ *
  *  @return
  *      zero or non-zero - Number of written bytes
  *      negative - I2C_ERROR_XXX status
  */
-int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop);
+int32_t i2c_write(i2c_t *obj, uint16_t address, const void *data,
+                  uint32_t length, bool stop);
 
-/** Reset I2C peripheral. TODO: The action here. Most of the implementation sends stop()
+/** Blocking reading data
  *
- *  @param obj The I2C object
- */
-void i2c_reset(i2c_t *obj);
-
-/** Read one byte
+ *  This function receives data, when the peripheral is configured as Master
+ *  from the selected slave, and when configured as Slave from the Master.
  *
- *  @param obj The I2C object
- *  @param last Acknoledge
- *  @return The read byte
- */
-int i2c_byte_read(i2c_t *obj, int last);
-
-/** Write one byte
+ *  This function is blocking, it will return when the transfer is complete or a
+ *  timeout event is triggered. The number of bytes received is returned by
+ *  the function after the operation is completed. Receive operation cannot be
+ *  cancelled or aborted.
  *
- *  @param obj The I2C object
- *  @param data Byte to be written
- *  @return 0 if NAK was received, 1 if ACK was received, 2 for timeout.
+ *  When in master mode the operation consists of:
+ *   - Address the slave as a Master receiver.
+ *   - Receive data from the addressed slave.
+ *   - Generate a STOP condition if the specified `stop` field is true.
+ *
+ *  @param obj     The I2C object
+ *  @param address 7-bit address (last bit is 1)
+ *  @param data    The buffer for receiving
+ *  @param length  Number of bytes to read
+ *  @param last    If true, indicates that the transfer contains the last byte
```

Comment:
This seems to be an equivalent to `stop` so shouldn't this be removed from the parameter list as per the design doc ?

Reply:
fixed